scrapy/settings/__init__.py

```diff
@@ -83,6 +83,7 @@ class BaseSettings(MutableMapping):
     def __init__(self, values=None, priority='project'):
         self.frozen = False
         self.attributes = {}
+        self._singletons = {}
```

Comment:
Is this a leftover from a previous implementation? There seems to be no other occurrence in this diff.

Reply:
Good catch!